src/main/java/Dog.java

```diff
@@ -38,8 +39,14 @@ public void doeat(){
     }
 
     public void dowalk(){
-        System.out.println(name+" is walikng");
-        this.walk=true;
+        if(this.eat==true){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the literal "true" boolean value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1125)